src/anemoi/training/train/forecaster.py

```diff
@@ -529,18 +529,11 @@ def calculate_val_metrics(
                 validation metrics and predictions
         """
         metrics = {}
-
         # Added to impute nans
-        nan_locations = torch.isnan(y[..., self.data_indices.internal_data.output.full])
+        nan_locations = torch.isnan(y)
```

Comment:
Can you explain this change please?

Reply:
This was throwing an IndexError, "self.data_indices.internal_data.output.full" had indices larger than the current y shape

Reply:
I am concerned that this is masking an issue with incompatible datasets.
Maybe this should be tested in conjunction with https://github.com/ecmwf/anemoi-training/pull/120

Reply:
Reverted this change ✔️